gateway-server/src/main/java/org/apache/knox/gateway/websockets/ProxyWebSocketAdapter.java

```diff
@@ -57,6 +61,12 @@
 
   private ExecutorService pool;
 
+  /* Message buffer for holding data frames temporarily in memory till connection is setup.
+   Keeping the max size of the buffer as 100 messages for now. */
+  private List<String> messageBuffer;
```

Comment:
You may consider instantiating the List here and check for emptiness in the below code instead of null checks. Less code and less error-prone.

Reply:
Done. Thanks